### PR TITLE
Add three new APIs to microsoft.resourcehealth

### PR DESCRIPTION
**Pull request by wengjn**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
Comment (KrisBash) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json]: The "child" token in these paths is out of spec with ARM contract. If this a singleton - and there will not be many child instances-, this is more appropriate:
`/{resourceUri}/providers/Microsoft.ResourceHealth/childAvailabilityStatuses/current`

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
Comment (KrisBash) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json]: Same as above. I'd recommend /childAvailabilityStatuses

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
Comment (KrisBash) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json]: I can ask for a second opinion, but I'd prefer the resource type name of "childResources" over "children." While "children" is grammatically plural, it feels inconsistent.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
Comment (KrisBash) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json]: One thing to consider is if this would be more future-proof as enum, such as resourceType or resourceManager with values of ARM or other. If you may want to distinguish between types of non-ARM resources in the future, it's more trivial to expand a string enum.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
Comment (KrisBash) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json]: I'm not seeing this definition in the swagger

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
Comment (KrisBash) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json]: I'm not seeing this definition in the swagger

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
Comment (KrisBash) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json]: I'm not seeing this definition in the swagger

**Review by KrisBash — CHANGES_REQUESTED** [review]
Changes needed from ARM

**Comment by KrisBash** [comment]
@wengjn feel free to set up a meeting with ARM API review if you'd like to discuss.

**Comment by ravbhatnagar** [comment]
@wengjn - please let us know when this is ready for re-review by ARM or feel free to set up a meeting.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
Comment (wengjn) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json]: Thanks for looking into this. this is the old commit which I have changed. I'm not adding new schema. I'm reusing the existing ones that we have.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
Comment (wengjn) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json]: Thanks for looking into this. this is the old commit which I have changed. I'm not adding new schema. I'm reusing the existing ones that we have.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
Comment (wengjn) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json]: Thanks for looking into this. this is the old commit which I have changed. I'm not adding new schema. I'm reusing the existing ones that we have.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
Comment (wengjn) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json]: Thanks very much for your suggestion. This flag would just stay as boolean as it only indicate to the client that whether to show further details for child resource. If it is arm type resource, we would be able to get more details. If not, client/caller would not be able to get child health status. That's it. As of now, I don't see necessary of making any future-proof on this property. :)

**Comment by wengjn** [comment]
This is updated. All comments have been resolved. Thx

**Review by KrisBash — APPROVED** [review]
LGTM after changes. Thanks

**Review by praries880 — APPROVED** [review]

**Pull request merged**
